allocator.go

```diff
@@ -78,6 +78,80 @@ type Allocator struct {
 	opts options
 }
 
+type NodeInterface struct{
+	n Node
+	NodeMap map[NodeID]Node
+}
+
+type RangeInterface struct {
+	r        Range
+	RangeMap map[RangeID]Range
+}
+
+type Cluster interface {
+	ClusterWriter
+}
+
+type ClusterWriter interface{
+	AddNode()
+	RemoveNode () bool
+	AddRange ()
+	RemoveRange () bool
+}
```

Comment:
I think we can remove the `ClusterWriter` + `ClusterReader` approach. I remember we talked about including this last time, but we're trying to ensure immutability within a package we own, and now I'm thinking it's not worth the overhead.